### PR TITLE
feat(core): Rate limit how frequently pubsub query messages can be published

### PR DESCRIPTION
**Pull request by stbrody**

### CODE REVIEW

File: packages/core/src/pubsub/__tests__/pubsub-ratelimit.test.ts [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    mockNow.mockReturnValueOnce(startTime)
+    mockNow.mockReturnValueOnce(new Date(startTime.getTime() + 100))
+    mockNow.mockReturnValueOnce(new Date(startTime.getTime() + 200))
+    mockNow.mockReturnValueOnce(new Date(startTime.getTime() + 300))
+    mockNow.mockReturnValueOnce(new Date(startTime.getTime() + 400))
+    mockNow.mockReturnValue(new Date(startTime.getTime() + 500))
+
+    let maxQueriesPerSecondPublished
+    let maxQueriesPerSecondPlusOnePublished
+    const allQueriesPublished = new Promise<void>((resolveAllQueriesPublished, _) => {
```
Comment (stbrody) [inline packages/core/src/pubsub/__tests__/pubsub-ratelimit.test.ts:107]: I really hate having to do these nested blocks just to use multiple promises at once.  Is there any nicer way to do this @ukstv?

**Comment by stbrody** [comment]
CC @smrz2001

**Review by oed — COMMENTED** [review]
lgtm, but want Sergey to review bc he understands observables better!

**Review by ukstv — APPROVED** [review]
🏇

**Pull request merged**